arch/arm/boot/dts/armada-385-ts7840.dts

```diff
@@ -0,0 +1,731 @@
+				address-width = <16>;
+				pagesize = <64>;
+			};
+		};
+
+		opencores_spi1: spi@420 {
+			compatible = "opencores,spi-oc";
+			reg = <0 0x420 32>;
+			interrupt-parent = <&fpgairqc>;
+			interrupts = <12 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&fpga_pll0c1>;
+			clock-names ="spi-oc-clk";
+			opencores-spi,idx = <0>;
+			opencores-spi,num-chipselects = <1>;
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			spidev@0 {
+				compatible = "spidev";
+				reg = <0>;
+				spi-max-frequency = <1000000>;
+			};
+		};
+
+		mikro_i2c: i2c@500 {
+			compatible = "opencores,i2c-ocores";
+			reg = <0 0x500 4>;
+			interrupt-parent = <&fpgairqc>;
+			interrupts = <13 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&fpga_pll0c1>;
+			clock-names = "i2c-oc-clk";
+		};
+
+		mikro_pwm: pwm@600 {
+			compatible = "technologic,pwm";
+			reg = <0 0x600 8>;
+			clocks = <&fpga_pll0c1>;
+			clock-names = "pwm-input-clk";
+		};
+
+		ts_mmc: mmc@800 {
+			compatible = "ts,tssdcore2";
+			reg = <0 0x800 32>;
+			//interrupt-parent = <&fpgairqc>;
+			//interrupts = <14 IRQ_TYPE_LEVEL_HIGH>;
+			busy-gpios = <&fpga_bank0 15 GPIO_ACTIVE_HIGH>;
+			debug-gpios = <&fpga_bank1 12 GPIO_ACTIVE_HIGH>;
+			status = "disabled";
+		};
+
+		mikro_adc0: adc@900 {
+			compatible = "technologic,ts-simple-adc";
+			reg = <0 0x900 4>;
+			interrupt-parent = <&fpgairqc>;
+			interrupts = <21 IRQ_TYPE_LEVEL_HIGH>;
+		};
+	};
+
+	sfp0: sfp@0 {
+		compatible = "sff,sfp";
+		i2c-bus = <&sfp0_i2c>;
+	};
+
+	sfp1: sfp@1 {
+		compatible = "sff,sfp";
+		i2c-bus = <&sfp1_i2c>;
+	};
+};
+
+&ahci0 {
+	status = "okay";
+};
+
+&bm {
+	status = "okay";
+};
+
+&bm_bppi {
+	status = "okay";
+};
+
+/* magjack/sfp port 0 */
+&eth0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ge0_rgmii_pins>;
+	status = "okay";
+	phy = <&phy0>;
+	phy-mode = "rgmii-id";
+	buffer-manager = <&bm>;
+	/*sfp = <&sfp0>;*/
+	bm,pool-long = <0>;
+	bm,pool-short = <1>;
+};
+
+/* magjack/sfgp port 1 */
+&eth1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ge1_rgmii_pins>;
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+	/*sfp = <&sfp1>;*/
+	bm,pool-long = <2>;
+	bm,pool-short = <1>;
+	buffer-manager = <&bm>;
+};
+
+/* sgmii port to eth switch */
+&eth2 {
+	status = "okay";
+	phy-mode = "sgmii";
+	bm,pool-long = <3>;
+	bm,pool-short = <1>;
+	buffer-manager = <&bm>;
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&gpio0 {
+	gpio-line-names = "", "", "I2C_0_CLK", "I2C_0_DAT", "", "", "", "", "",
+		"", "", "", "", "", "", "", "", "", "WIFI_IRQ_N", "FPGA_RESET",
+		"", "", "", "", "", "", "", "", "", "", "", "";
+};
+
+&gpio1 {
+	gpio-line-names = "", "CPU_SPEED_0", "CPU_SPEED_3", "CPU_SPEED_4",
+		"CPU_TYPE_0", "", "", "", "", "", "GPS_PPS", "ETH_SW_INT#",
+		"CPU_TYPE_1", "", "", "", "", "FPGA_FLASH_SELECT", "EMMC_CMD",
+		"SPREAD_SPECTRUM#", "", "DETECT_9478", "", "", "", "", "", "",
+		"", "", "", "";
+};
+
+&i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c0_pins>;
+	clock-frequency = <100000>;
```

Comment:
I pointed this out on the 7800v2 DTS but missed it here. Is there any reason this bus is limited to 100 kHz?

Reply:
Leaving at 100khz since the microcontroller may not support 400khz.